lib/NativeCode/DynamicLibraryLoaderHelper/NativeRender/include/Config/SteamConfig.hpp

```diff
@@ -0,0 +1,146 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#pragma once
+
+#include "config_legacy.h"
+#include "include/Config/Config.hpp"
+
+namespace pew::eos::config
+{
+    struct SteamConfig final : Config
+    {
+        /**
+         * \brief Major version of the steam SDK used.
+         */
+        uint32_t steam_sdk_major_version;
+
+        /**
+         * \brief Minor version of the steam SDK used.
+         */
+        uint32_t steam_sdk_minor_version;
+
+        /**
+         * \brief Any platform management flags that need setting because of
+         * Steam integration. Defaults to none.
+         */
+        EOS_EIntegratedPlatformManagementFlags integrated_platform_management_flags = EOS_EIntegratedPlatformManagementFlags::EOS_IPMF_Disabled;
+
+        bool try_get_library_path(std::filesystem::path& library_path) const
+        {
+            if (exists(_override_library_path))
+            {
+                library_path = absolute(_override_library_path);
+                return true;
+            }
+
+            if (exists(_library_path))
+            {
+                library_path = absolute(_library_path);
+                return true;
+            }
+
+            return false;
+        }
+
+        
+        const char* get_steam_api_interface_versions_array() const
+        {
+            // For each element in the array (each of which is a string of an api version information)
+            // iterate across each character, and at the end of a string add a null terminator \0
+            // then add one more null terminator at the end of the array
+            std::vector<char> steamApiInterfaceVersionsAsCharArray;
+
+            for (const auto& currentFullValue : _steam_api_interface_versions_array)
+            {
+                for (char currentCharacter : currentFullValue)
+                {
+                    steamApiInterfaceVersionsAsCharArray.push_back(currentCharacter);
+                }
+
+                steamApiInterfaceVersionsAsCharArray.push_back('\0');
+            }
+            steamApiInterfaceVersionsAsCharArray.push_back('\0');
+
+            return steamApiInterfaceVersionsAsCharArray.data();
```

Comment:
It _is_ significant! And good catch. I updated the implementation so that instead of returning a `const char*` that was allocated on the stack (and is therefore not guaranteed to be valid after returning), I changed the function to return a `std::string` value, which is handled appropriately. 

The place where the return value of this function is applied to the EOS SDK was also updated to append the `.c_str()` method so that when the value is assigned it's done correctly.

Reply:
Not a required change: I believe since C++11 you can use `string::insert` to add a range of chars. So if you change `steamApiInterfaceVersionsAsCharArray` from a `std::vector<char>` to a `std::string` I believe you should be able to change the code to something like this, and avoid the copy or potentially as many allocations (depending on how often `push_back` causes an alloc to occur): 

```
  std::string steamApiInterfaceVersionsAsCharArray;

  for (const auto& currentFullValue : _steam_api_interface_versions_array) {
     steamApiInterfaceVersionsAsCharArray.insert(steamApiInterfaceVersionsAsCharArray.end,
                                                                              currentFullValue.begin(),
                                                                              currentFullValue.end()
     );
     steamApiInterfaceVersionsAsCharArray.push_back('\0');
}
steamApiInterfaceVersionsAsCharArray.push_back('\0');

return steamApiInterfaceVersionsAsCharArray.data();
```

Reply:
#EOS-2279